include/zephyr/platform/hooks.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_PLATFORM_PLATFORM_H_
+#define ZEPHYR_INCLUDE_PLATFORM_PLATFORM_H_
+
+/**
+ * @file
+ * @brief Platform hooks
+ *
+ * This header file contains function prototypes for the interfaces between
+ * zephyr architecture and initialization code and the platform pecific logic
+ * that resides under boards/ and soc/
+ *
+ * @note These are all standard platform interfaces that are exported from
+ * platform specific logic to OS internal logic.  These should never be accessed
+ * directly from application code but may be freely used within the OS.
+ */
+
+
+/**
+ * @brief Platform hook executed at the beginning of the reset vector.
```

Comment:
```suggestion
 * @brief SoC hook executed at the beginning of the reset vector.
```

Same change should be done for the remaining SoC hooks.